blockchain/contracts/StakingNFT.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StakingContract is IERC721Receiver {
+    IERC721 public nft;
```

Comment:
Give this a more descriptive name - consider `characterToken`

Reply:
noted, will do